Add Sueños Nocturnos StoryQuest

xXLeopoldoXx opened the pull request:
Cuando su hermana desaparece misteriosamente durante el sueño, Tim se lanza a un mundo de pesadillas vivientes donde la realidad se retuerce con sus temores más profundos. Enfrentando un laberinto que devora la memoria, una mazmorra poseída por melodías espectrales y un dios del inframundo sediento de almas, Tim deberá superar pruebas imposibles para reunir los fragmentos de luz que lo guíen hasta ella... antes de que el sueño se convierta en prisión eterna.

//

When his sister mysteriously disappears during his sleep, Tim is thrust into a world of living nightmares where reality twists with his deepest fears. Facing a memory-devouring labyrinth, a dungeon haunted by spectral melodies, and an underworld god thirsting for souls, Tim must overcome impossible trials to gather the shards of light that will guide him to her—before sleep becomes an eternal prison.

Art:

- PjIddle.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- elfo_iddle.png – CC-BY-SA 4.0 – por @xXLeopoldoXx y @WhoD-S
- angel_checkponit.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- llave 1.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- Fantasma_idle_V7_(final).png – CC-BY-SA 4.0 – por @Leonardo-UTP
- minigame1boss.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- niña64x64_estesi.png – CC-BY-SA 4.0 – por @GianPierooo
- shjourney_Dragon.png – CC-BY-SA 4.0 – por @WhoD-S
- shjourney_Eyes.png – CC-BY-SA 4.0 – por @WhoD-S
- shjourney_Ghost.png – CC-BY-SA 4.0 – por @WhoD-S
- shjourney_Guardian.png – CC-BY-SA 4.0 – por @WhoD-S
- shjourney_Pillar.png – CC-BY-SA 4.0 – por @WhoD-S
- BallFire.png - CC-BY-SA 4.0 – por @xXLeopoldoXx
- BallFire2.png  (green) – CC-BY-SA 4.0 – por @GianPierooo
- heart_pixelart_64x64 – CC-BY-SA 4.0 – por @xXLeopoldoXx
- tilesetCombate final – CC-BY-SA 4.0 – por @xXLeopoldoXx
- DemonSupay – CC-BY-SA 4.0 – por @xXLeopoldoXx
- fat_ghost_4 – CC-BY-SA 4.0 – por @Leonardo-UTP
- mapa3 and presentacion (sprites) – CC-BY-SA 4.0 – por @Leonardo-UTP
- Tree_ojos.png – CC-BY-SA 4.0 – por @GianPierooo
- Elfo del bosque.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- AnimationSheet_Character.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- Modelo_PJ_Walking.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- Modelo_PJ_running.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- Modelo_protaBoceto-SheetIddle.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- bloque_negro v2.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- baños1.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- fantasma_discipulo_taemanto_2_.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- fantasma_discipulo_taemanto_idle.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- scream-90747.wav – CC-BY-SA 4.0 – por @Leonardo-UTP
- escritorio1.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- pisomadera1.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- pisomadera2.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- techobaño.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- muro1.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- fantasma1(recolor)v5.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- Fantasma_idle_V7_(final).png – CC-BY-SA 4.0 – por @Leonardo-UTP
- fat_ghost_1_angy_v3(perfect) 100.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- fat_ghost_4_idle_(perfect).png – CC-BY-SA 4.0 – por @Leonardo-UTP
- image.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- jefe1boceto2.5sinPIEZAS.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- magenmader1.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- floresazulas1.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- angel1puerta.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- angel1puerta2.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- angel1puerta3(perfect).png – CC-BY-SA 4.0 – por @Leonardo-UTP
- fuente1.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- piedras2.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- piedras3.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- espada_sprite.png – CC-BY-SA 4.0 – por @GianPierooo
- DemonFinal-Sheet.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- DemonFinal-SheetAnticipation.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- DemonFinalAttack.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- heart_pixelart_64x64_clean_transparent.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- DemonFinal-SheetAnticipationBigger.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- DemonFinal-SheetAnticipationBigger2.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- Dark-grass-tileset.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- GrassHills.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- Rocks_04.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- Tree.png – CC-BY-SA 4.0 – por @Leonardo-UTP
- Modelo protaBoceto static1.png – CC-BY-SA 4.0 – por @xXLeopoldoXx
- niña.png – CC-BY-SA 4.0 – por @GianPierooo
- niña64x64.png – CC-BY-SA 4.0 – por @GianPierooo
- eeeeeeeee.png – CC-BY-SA 4.0 – por @GianPierooo
- niña.pngPjIddle.png – CC-BY-SA 4.0 – por @GianPierooo

Music:

-INTRO-SHADOWJOURNEY.ogv – CC-BY-SA 4.0 – por @xXLeopoldoXx
- Other third-party assets as indicated in their `.license` files


Programming:

- Intro1 level by @xXLeopoldoXx
- IntroAnimation level by @xXLeopoldoXx
- Intro02 level by @xXLeopoldoXx
- Minigame3Combat level by @xXLeopoldoXx
- IntroLabyrinth1 level by @GianPierooo
- Outro level by @GianPierooo
- Labyrinth4 level by @Leonardo-UTP
- Minigame2 Musical Puzzle level by @WhoD-S

 Game Design:

- Core gameplay and mechanics by @Leonardo-UTP and @WhoD-S
- Map and character design by @xXLeopoldoXx
- Puzzles, enemies, and narrative progression by the whole team

Resolves https://github.com/endlessm/threadbare/issues/1079

manuq commented:
@xXLeopoldoXx @WhoD-S @Leonardo-UTP @GianPierooo muchísimas gracias por ofrecer esta StoryQuest para Threadbare! Como me habían comentado, veo que hay cambios radicales en elementos del juego. Como llaves que el jugador puede obtener.  Deberían llevar la cuenta de las llaves en la escena del nivel que está dentro de la carpeta de la storyquest, no en el player:
```
scenes/quests/story_quests/shjourney/4_Laberinto/llave.gd

   7   │ +extends Sprite2D
   8   │ +
   9   │ +func CuandoEntraJugador(body: Node2D) -> void:
  10   │ +   if body.is_in_group("player"):
  11   │ +       body.llaves += 1
```

Veo que han cambiado el código de player para que reproduzca una animación distinta al correr. El juego actual reproduce la misma animación de caminar al doble de velocidad. Para esto, no veo alternativa, habría que descartar el cambio y elegir con cuál de las animaciones se quieren quedar. Que camine todo el tiempo o que corra todo el tiempo:
```
scenes/game_elements/characters/player/components/animation_player.gd

  23   │ -   var double_speed: bool = current_animation == &"walk" and player.is_running()
  24   │ -   speed_scale = original_speed_scale * (2.0 if double_speed else 1.0)
  25   │ -
  26   │ -
  27   │  func _process_walk_idle(_delta: float) -> void:
  28   │     if player.velocity.is_zero_approx():
  29   │         play(&"idle")
  30   │ +   elif player.is_running():
  31   │ +       if has_animation("sprint"):
  32   │ +           play(&"sprint")
  33   │ +       else:
  34   │ +           play(&"walk")
  35   │     else:
  36   │         play(&"walk")
```

Ahora bien, siendo tantos los cambios que hay que hacer y habiendo trabajo que tendrían que descartar (como la animación de correr del personaje), me pregunto: ¿Realmente quieren hacerlo? ¿No se merece Sueños nocturnos ser un juego aparte, separado de Threadbare?

----

@xXLeopoldoXx @WhoD-S @Leonardo-UTP @GianPierooo thank you so much for offering this StoryQuest for Threadbare! As you told me, I see some radical changes to elements of the game, like keys the player can obtain. You should keep track of the keys in the level scene inside the StoryQuest folder, not in the player:
```
scenes/quests/story_quests/shjourney/4_Laberinto/llave.gd

   7   │ +extends Sprite2D
   8   │ +
   9   │ +func CuandoEntraJugador(body: Node2D) -> void:
  10   │ +   if body.is_in_group("player"):
  11   │ +       body.llaves += 1
```

I see that you have changed the player code to play a different animation when running. The current game plays the same animation for walking at twice the speed. I don't see any alternative to this; you'll have to discard the change and choose which animation they want to keep. Either walk all the time or run all the time:

```
scenes/game_elements/characters/player/components/animation_player.gd

  23   │ -   var double_speed: bool = current_animation == &"walk" and player.is_running()
  24   │ -   speed_scale = original_speed_scale * (2.0 if double_speed else 1.0)
  25   │ -
  26   │ -
  27   │  func _process_walk_idle(_delta: float) -> void:
  28   │     if player.velocity.is_zero_approx():
  29   │         play(&"idle")
  30   │ +   elif player.is_running():
  31   │ +       if has_animation("sprint"):
  32   │ +           play(&"sprint")
  33   │ +       else:
  34   │ +           play(&"walk")
  35   │     else:
  36   │         play(&"walk")
```

Now, with so many changes to be made and work that would have to be dropped (like the character's running animation), I wonder: Do you really want to do this? Doesn't Sueños Nocturnos deserve to be your own game, separate from Threadbare?

manuq commented:
(English below)

@xXLeopoldoXx ¡excelente! Me alegra saber que quieran integrar Sueños nocturnos. Va a ser una gran contribución a Threadbare! Voy a hacer que se pueda cambiar el sprite al correr, así no tienen que descartar eso. Mientras tanto, ustedes podrían avanzar en:
- Hacer que las 3 llaves se cuenten en el nivel, no en el personaje player.
- Quitar todos los assets que no se utilizan. Por ejemplo si agregaron todo un asset pack para usar un árbol, borren todos los demás assets excepto el del árbol. Si agregaron una escena pero después no la usaron, elimínenla.

Espero que al hacer esto último la página de GitHub no se trabe y podamos revisar!

----

@xXLeopoldoXx excellent! I'm glad you want to integrate Sueños nocturnos. It's going to be a great contribution to Threadbare! I'm going to make it possible to change the sprite while running, so you don't have to discard that. In the meantime, you could work on:
- Making the 3 keys count happen in the level, not in the player character.
- Removing all unused assets. For example, if you added an entire asset pack to use a tree, delete all the other assets except the tree. If you added a scene but didn't use it afterward, delete it.

I hope doing this last step will keep the GitHub page from crashing so we can review!

manuq commented:
@wjt in my merge PR I see:

```
Encountered 1 file that should have been a pointer, but wasn't:
	scenes/quests/story_quests/shjourney/0_IntroVideo/INTRO-SHADOWJOURNEY.ogv
```

I guess this is because the rule for OGV was added after. So if I do a rebase instead of a merge the pointer will go away, I guess?

In any case this video is fading in/out lines of text. Ideally it should be reimplemented in Godot with labels animated with an AnimationPlayer node.

manuq commented:
(English below)

@xXLeopoldoXx nos encontramos con el siguiente problema de licencias en los assets de terceros:

> pixel-art-top-down-basic – CC-BY-SA 4.0 – por cainos – obtenido de https://cainos.itch.io/pixel-art-top-down-basic

Yendo a la página de itchio, no veo que la licencia sea "CC-BY-SA 4.0 ". Por el contrario, dice "You may not redistribute it or resell it.".

> pixel-rooms – CC-BY-SA 4.0 – por ToffeeCraft – obtenido de https://toffeecraft.itch.io/pixel-rooms

Lo mismo con este. No dice "CC-BY-SA 4.0" en ningún lado, y dice "Not redistribute or resell this assets."

La única alternativa que veo es que reemplacen los gráficos con arte original. ¿Creen que pueden hacerlo?

----

@xXLeopoldoXx we found the following licensing issues with third-party assets:

> pixel-art-top-down-basic – CC-BY-SA 4.0 – por cainos – obtenido de https://cainos.itch.io/pixel-art-top-down-basic

Going to the itchio page, I don't see the license as "CC-BY-SA 4.0." On the contrary, it says "You may not redistribute or resell it."

> pixel-rooms – CC-BY-SA 4.0 – por ToffeeCraft – obtenido de https://toffeecraft.itch.io/pixel-rooms

Same with this one. It doesn't say "CC-BY-SA 4.0" anywhere, and it says "Do not redistribute or resell this asset."

The only alternative I see is to replace the graphics with original art. Do you think you can do that?

xXLeopoldoXx commented:
Hola @manuq claro que si, vamos a reemplazar esos assets donde hay problemas con licencia. El de cainos fue usado en grande parte de los minijuegos (pasto y árboles). Por ello, nos va a tomar un poco de tiempo en hacer otros, pero lo vamos a intentar.

manuq commented:
@xXLeopoldoXx here I did another merge of latest Threadbare that resolves the conflict above: https://github.com/UTPGameLab4-0-Grupo6-Equipo1/threadbare/pull/97

manuq commented:
Hi team! How are you going with the cleanup? Please remember that everything should go in your StoryQuest folder. I still see 26 files changed outside `scenes/quests/story_quests/shjourney`:

```
$ git diff main... --stat ':!scenes/quests/story_quests/shjourney' 
 build/.gitkeep                                                                                                     |   0
 build/web/.gitkeep                                                                                                 |   0
 project.godot                                                                                                      |   1 +
 scenes/game_elements/characters/enemies/guard/guard.tscn                                                           |  60 ++++++++++++++++++++++----------------------
 scenes/game_elements/characters/player/components/player.gd                                                        |   1 +
 scenes/game_elements/characters/player/components/player_fighting.gd                                               |  17 ++++++++-----
 scenes/game_elements/characters/player/components/player_interaction.gd                                            |  30 +++++++++++++++++-----
 scenes/game_elements/props/checkpoint/checkpoint.tscn                                                              |  12 ++++++---
 scenes/game_elements/props/collectible_item/collectible_item.tscn                                                  |   2 +-
 scenes/game_elements/props/collectible_item/components/collectible_itemMinijuego2.gd                               | 109 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 scenes/game_elements/props/collectible_item/components/collectible_itemMinijuego2.gd.uid                           |   1 +
 scenes/game_elements/props/decoration/bush/bush.tscn                                                               |   2 +-
 scenes/game_elements/props/door/components/door.gd                                                                 |  31 ++++++++++++++++++++++-
 scenes/game_elements/props/door/door.tscn                                                                          |  17 ++++++++++---
 scenes/game_elements/props/sequence_puzzle_hint_sign/sequence_puzzle_hint_sign.tscn                                |   1 +
 scenes/game_elements/props/tree/components/tree_spriteframes_green.tres                                            |  12 ++++-----
 scenes/game_elements/props/tree/tree.tscn                                                                          |   1 +
 scenes/game_logic/fill_game_logic.gd                                                                               |  51 +++----------------------------------
 scenes/quests/story_quests/template/0_template_intro/template_intro_components/template_intro.dialogue             |  22 +++++++++++-----
 scenes/quests/story_quests/template/2_template_combat/template_combat_components/template_combat.dialogue          |  11 +++++---
 scenes/quests/story_quests/template/2_template_combat/template_combat_components/template_target_spriteframes.tres |  43 +++++++++++++++++++++++--------
 scenes/quests/story_quests/template/2_template_combat/template_combat_components/template_throwing_enemy.tres      | 154 +++++++++++++++++++++++++++++++++++++++-------------------------------------------------------------------------
 scenes/quests/story_quests/template/3_template_sequence_puzzle/template_sequence_puzzle.dialogue                   |   8 ------
 scenes/quests/story_quests/template/3_template_sequence_puzzle/template_sign.tres                                  |  75 +++++++++++++++++++++++--------------------------------
 scenes/ui_elements/cinematic/cinematic.gd                                                                          |   4 ++-
 scenes/ui_elements/cinematic/cinematicVideo.gd.uid                                                                 |   1 +
 26 files changed, 386 insertions(+), 280 deletions(-)
```

manuq commented:
The team said through another channel (Discord) that this is ready for a new review.

manuq commented:
@xXLeopoldoXx please check these set of changes I did: https://github.com/UTPGameLab4-0-Grupo6-Equipo1/threadbare/pull/118

xXLeopoldoXx commented:
> @xXLeopoldoXx et all, is the list of artwork in the ticket description up to date? I see a lot of assets in the pull request.

Hi @manuq , we've already updated the latest assets we added in the description. That's all for now.

wjt commented:
Please also take a look at:

- https://github.com/UTPGameLab4-0-Grupo6-Equipo1/threadbare/pull/120

wjt commented:
> Hello @manuq @wjt , we would like to know if we can add AI-created sounds to replace minor sounds like a lamp turning on, enemies laughing, etc. It's worth noting that all ambient music for each scene has been replaced and unused music has been removed.

https://github.com/endlessm/threadbare/blob/main/docs/LICENSING.md#ai-generated-assets-and-code documents our policy on AI-created assets.

xXLeopoldoXx commented:
Hello @manuq and @wjt , we have already removed the previous sound effects, there is no longer any AI in the StoryQuest and all the sound effects have been taken from https://freesound.org all with a Creative Commons 0 license. We are waiting for further instructions

wjt commented:
@xXLeopoldoXx In the sidebar of this pull request, do you see a checkbox that says:

> - [ ] Allow edits and access to secrets by maintainers

If so, can you enable it? I have some fixes I would ideally like to push back to this PR.

<img width="317" height="974" alt="image" src="https://github.com/user-attachments/assets/b34d1686-0ac7-4082-9674-54e9b8ff6305" />

xXLeopoldoXx commented:
> Thanks for checking. I am not sure why it doesn't appear for some PRs…
> 
> @Stregoica777 has just added me to your organisation so that I can push directly

Perfect, we'll be on the lookout for anything.

wjt commented on scenes/quests/story_quests/shjourney/3_shjourney_intro/template_intro0.2.tscn:
```diff
@@ -0,0 +1,722 @@
+point_count = 2
+
+[sub_resource type="AudioStreamMP3" id="AudioStreamMP3_oixsc"]
+data = PackedByteArray("SUQzBAAAAAAAI1RTU0UAAAAPAAADTGF2ZjU5LjI3LjEwMAAAAAAAAAAAAAAA//u0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAASW5mbwAAAA8AAAC5AAGigAAEBgkMDxETFhgbHiEjJScqLTAyNTc5PD9CREZIS05RU1ZYWl1gY2VoaWxvcnR3eXt+gYSGiYuNkJOWmJucn6Klp6qsrrG0t7m8vcDDxsjLzs/S1dja3d7h5Ofp7O/w8/b5+/4AAAAATGF2YzU5LjM3AAAAAAAAAAAAAAAAJAVAAAAAAAABooD97nQ7AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA ...
```
Here you have embedded an MP3 file directly into the scene. Please save it as a separate `.mp3` file and reference that in the scene.

wjt commented on scenes/quests/story_quests/shjourney/Otros_componentes/Enemigos/fantasmas/fantasmas.tscn:
```diff
@@ -0,0 +1,301 @@
+}
+
+[sub_resource type="AudioStreamMP3" id="AudioStreamMP3_wyebm"]
+data = PackedByteArray("//vURAAABLQ80QVh4ACd6xoQrCQAWClfbbmMAAsCLO0zMYABLSF4FqPiDQmdZtWCnsaM6wE9qSVBZQtonRFVzgLAKgsEFD0PZ5E4aZpnWo1er2ePu973vDfsavVicQxDFZE937yJl+r2d/ZgQ9D2ePAYFZEhqxWPKf/0vDfv37+Ph4r1er37++/S99//+9KU94b/dH79/cP/gg4EDgkBA5/5QHygIBgSAg6oHw+UBCoEAxBAEDkHAQph8xALQLUZ+CRmt5xeApteAIy37pF/C2iPjqTb/gADDJGGxW3UJznPf4QRo0aOdZaMUAgAAAAGCTF0YoYTFYJituiAEwTFbdIBQxYoJGLRo0bdI0aNGjRtyQCsVitGjbn ...
```
Another embedded MP3 - please save this as a separate `.mp3` file.

wjt commented:
Thank you for your hard work and continued support on asset licensing. We are so close now!

wjt reviewed (approved):
I think this is ready to merge! Well done for all your hard work and patience working through asset licensing!

I know that you found some bugs, probably introduced by other changes made to the game in parallel. Please, continue to contribute further pull requests to improve your StoryQuest and fix bugs that you find.

wjt commented:
Now that your storyquest is merged, please send fixes and improvements for it by opening pull requests directly to this fork (`endlessm:main`) rather than to your own fork! I will open one in a moment and seek your review.

wjt commented:
I have invited the contributors here to a team in our GitHub organisation @endlessm/suenos-nocturnos. In #1293 I am marking that team as the co-maintainer of this quest. This will give you the ability to merge changes to your own quest in this repository, as well as pushing branches to this repo rather than your fork, as you make any further changes to the quest. There's no obligation for you to accept the invitation! But we'd be delighted to have you.

xXLeopoldoXx commented:
We are very happy that our StoryQuest has finally joined, we will be looking forward to fixing the bugs it has and adapting it to the new Threadbare updates, thank you very much @wjt @manuq
